sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -1319,39 +1320,34 @@ def test_enet_cv_sample_weight_correctness(fit_intercept, sparse_container):
         X = sparse_container(X)
     params = dict(tol=1e-6)
 
-    # Set alphas, otherwise the two cv models might use different ones.
-    if fit_intercept:
-        alphas = np.linspace(0.001, 0.01, num=91)
-    else:
-        alphas = np.linspace(0.01, 0.1, num=91)
-
-    # We weight the first fold 2 times more.
-    sw[:n_samples] = 2
+    # We weight the first fold n times more.
+    sw[:n_samples] = rng.randint(0, 5, size=sw[:n_samples].shape[0])
     groups_sw = np.r_[
         np.full(n_samples, 0), np.full(n_samples, 1), np.full(n_samples, 2)
     ]
     splits_sw = list(LeaveOneGroupOut().split(X, groups=groups_sw))
-    reg_sw = ElasticNetCV(
-        alphas=alphas, cv=splits_sw, fit_intercept=fit_intercept, **params
-    )
+    reg_sw = ElasticNetCV(cv=splits_sw, fit_intercept=fit_intercept, **params)
     reg_sw.fit(X, y, sample_weight=sw)
 
     # We repeat the first fold 2 times and provide splits ourselves
     if sparse_container is not None:
         X = X.toarray()
-    X = np.r_[X[:n_samples], X]
+    X_rep = np.repeat(X, sw.astype(int), axis=0)
+    ##Need to know number of repitions made in total
+    n_reps = X_rep.shape[0] - X.shape[0]
+    X = X_rep
```

Comment:
I would rather not rename change the `X` variable to keep the code easier to follow.

Maybe you could instead name the variables `X_with_weights`, `y_with_weights`, `groups_with_weights` on one hand and `X_with_repetitions`, `y_with_repetitions` and `groups_with_repetitions` on the other hand.

Reply:
And similarly for the names of the 2 cross-validation splitters (if you adapt the code to use metadata routing) or the results of their splits if you prefer to precompute them instead of leveraging metadata routing.